source/IpcServer.cpp

```diff
@@ -0,0 +1,74 @@
+#include "IpcServer.hpp"
+#include "Server.hpp"
+#include "Output.hpp"
+
+#include <iostream>
+#include <unistd.h>
+
+IpcServer::IpcServer(std::string const& socket, Server *server)
+  : ipcServer(socket), socket(socket), server(server), thread(&IpcServer::accept, this)
+{
+}
+
+void IpcServer::accept()
+{
+  try {
+    while (1)
```

Comment:
Why the `while (1)` ? You have an infinite loop that breaks on exception inside it so it will only be executed once. A scope would have the same result.